modules/core/src/encoder/SemanticPreEncoder.ts

```diff
@@ -53,75 +53,89 @@ export class SemanticPreEncoder {
 
             } else if (vendor[gvlVendorKey].length === 0) {
 
-              /**
-               * If the vendor does exist, but they haven't declared any
-               * purposes for this legal basis, then we need to see if they can
-               * possibly have the legal basis from their flexible purposes.
-               * This, of course, only matters if it is a globally-scoped string.
-               */
-
-              if (tcModel.isServiceSpecific) {
+              if (
+                gvlVendorKey === 'legIntPurposes' && vendor['purposes'].length === 0 && vendor['legIntPurposes'].length === 0 && vendor['specialPurposes'].length > 0
+              ) {
 
-                if (vendor.flexiblePurposes.length === 0) {
+                /**
+                 * Per June 2021 Policy change, Vendors declaring only Special Purposes may
+                 * have theirlegitimate interest Vendor bit set if they have been disclosed. 
+                 * This empty block ensures their LI bit remains set
```

Comment:
We could avoid the "empty block" by checking the NOT of the condition and moving the `else` code into the "then". But, a big comment in an empty block is also good documentation, so up to you which is clearer.

Reply:
I think is crealer the comment as documentation